### PR TITLE
feature: Add WETH position manager

### PR DESCRIPTION
**Pull request by StefanVitor**

### CODE REVIEW

File: contracts/Interfaces/IPositionManagerWETH.sol [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import { ERC20PermitSignature } from "@tempusfinance/tempus-utils/contracts/utils/PermitHelper.sol";
+import { IWETH } from "../Dependencies/IWETH.sol";
+import { IPositionManagerDependent } from "./IPositionManagerDependent.sol";
+
+/// @notice Interface for the PositionManagerWETH contract.
+interface IPositionManagerWETH is IPositionManagerDependent {
+    // --- Events ---
+
+    /// @dev The position is changed using stETH.
```
Comment (iamvukasin) [inline contracts/Interfaces/IPositionManagerWETH.sol:12]:
```suggestion
    /// @dev The position is changed using ETH.
```

File: contracts/Interfaces/IPositionManagerWETH.sol [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    /// @param isDebtIncrease Whether the debt is added to the position or removed from it.
+    event ETHPositionChanged(
+        address indexed position,
+        uint256 collateralAmount,
+        bool isCollateralIncrease,
+        uint256 debtAmount,
+        bool isDebtIncrease
+    );
+
+    // --- Errors ---
```
Comment (iamvukasin) [inline contracts/Interfaces/IPositionManagerWETH.sol:26]:
```suggestion
    // --- Errors ---

```

File: contracts/Interfaces/IPositionManagerWETH.sol [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    /// @dev Sending Ether failed.
+    error SendingEtherFailed();
+
+    // --- Functions ---
+
+    /// @dev Returns the WETH address.
+    function wETH() external view returns (IWETH);
+
+    /// @dev Manages the position with stETH for the position (the caller).
+    /// @param collateralChange The amount of stETH to add or remove.
```
Comment (iamvukasin) [inline contracts/Interfaces/IPositionManagerWETH.sol:39]:
```suggestion
    /// @dev Manages the position with ETH for the position (the caller).
    /// @param collateralChange The amount of ETH to add or remove.
```

File: contracts/PositionManagerWETH.sol [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            isCollateralIncrease,
+            debtChange,
+            isDebtIncrease,
+            maxFeePercentage,
+            emptySignature
+        );
+
+        if (!isCollateralIncrease && collateralChange > 0) {
+            wETH.withdraw(collateralChange);
+            (bool success,) = msg.sender.call{ value: collateralChange }("");
```
Comment (yuval-weiss) [inline contracts/PositionManagerWETH.sol:76]: use `transfer` instead of call?

Comment (StefanVitor) [reply contracts/PositionManagerWETH.sol:76]: Why, `call` is recommended?

**Pull request merged**